apps/astarte_trigger_engine/lib/astarte_trigger_engine/config.ex

```diff
@@ -114,6 +114,13 @@ defmodule Astarte.TriggerEngine.Config do
     binding_skip: [:system],
     default: Astarte.TriggerEngine.EventsConsumer
 
+  @envdoc "The module used to keep track of the number of redeliveries, used for tests with Mox"
+  app_env :retry_counter, :astarte_trigger_engine, :retry_counter,
+    os_env: "TRIGGER_ENGINE_RETRY_COUNTER",
+    type: :module,
+    binding_skip: [:system],
+    default: Astarte.TriggerEngine.DeliveryRetryCounter
```

Comment:
The `binding_skip: [:system]` part allows to remove the variable binding from system configuration, isn't that enough?

Reply:
Reference for future self: Skogsra allows for runtime config changes, but this is a static (compile-time) parameter, so it really makes no sense to put it in there instead of `config/test.exs`